src/containers/Tenant/Query/utils/useSavedQueries.tsx

```diff
@@ -7,7 +9,9 @@ export function useSavedQueries() {
     const [savedQueries] = useSetting<SavedQuery[]>(SAVED_QUERIES_KEY, []);
     const filter = useTypedSelector(selectSavedQueriesFilter).toLowerCase();
 
-    return filter
-        ? savedQueries.filter((item) => item.body.toLowerCase().includes(filter))
-        : savedQueries;
+    const filteredSavedQueries = React.useMemo(() => {
+        return savedQueries.filter((item) => item.body.toLowerCase().includes(filter));
+    }, [savedQueries, filter]);
+
+    return filter ? filteredSavedQueries : savedQueries;
```

Comment:
if filter by default would be empty string, we may always use `filteredSavedQueries`